core/src/test/scala/com/databricks/labs/remorph/parsers/tsql/TSqlAstBuilderSpec.scala

```diff
@@ -33,4 +34,78 @@ class TSqlAstBuilderSpec extends AnyWordSpec with TSqlParserTestCommon with Matc
+
+    example(
+      query = "SELECT a - b FROM dbo.table_x",
+      expectedAst =
+        Project(NamedTable("dbo.table_x", Map.empty, is_streaming = false), Seq(Subtract(Column("a"), Column("b")))))
+
+    example(
+      query = "SELECT a * b FROM dbo.table_x",
+      expectedAst =
+        Project(NamedTable("dbo.table_x", Map.empty, is_streaming = false), Seq(Multiply(Column("a"), Column("b")))))
+
+    example(
+      query = "SELECT a / b FROM dbo.table_x",
+      expectedAst =
+        Project(NamedTable("dbo.table_x", Map.empty, is_streaming = false), Seq(Divide(Column("a"), Column("b")))))
+
+    example(
+      query = "SELECT a || b FROM dbo.table_x",
+      expectedAst =
+        Project(NamedTable("dbo.table_x", Map.empty, is_streaming = false), Seq(Concat(Column("a"), Column("b")))))
+
+    example(query = "SELECT a % b", expectedAst = Project(NoTable(), Seq(Mod(Column("a"), Column("b")))))
+
+    example(query = "SELECT a & b", expectedAst = Project(NoTable(), Seq(BitwiseAnd(Column("a"), Column("b")))))
+
+    example(query = "SELECT a | b", expectedAst = Project(NoTable(), Seq(BitwiseOr(Column("a"), Column("b")))))
+
+    example(query = "SELECT a ^ b", expectedAst = Project(NoTable(), Seq(BitwiseXor(Column("a"), Column("b")))))
+  }
+  "translate queries with complex expressions, including parens" in {
+    example(
+      query = "SELECT a + b * 2",
+      expectedAst = Project(NoTable(), Seq(Add(Column("a"), Multiply(Column("b"), Literal(integer = Some(2)))))))
+
+    example(
+      query = "SELECT (a + b) * 2",
+      expectedAst = Project(NoTable(), Seq(Multiply(Add(Column("a"), Column("b")), Literal(integer = Some(2))))))
+
+    example(
+      query = "SELECT a & b | c",
+      expectedAst = Project(NoTable(), Seq(BitwiseOr(BitwiseAnd(Column("a"), Column("b")), Column("c")))))
+
+    example(
+      query = "SELECT (a & b) | c",
+      expectedAst = Project(NoTable(), Seq(BitwiseOr(BitwiseAnd(Column("a"), Column("b")), Column("c")))))
+
+    example(
+      query = "SELECT a % 3 + b * 2 - c / 5",
+      expectedAst = Project(
+        NoTable(),
+        Seq(
+          Subtract(
+            Add(Mod(Column("a"), Literal(integer = Some(3))), Multiply(Column("b"), Literal(integer = Some(2)))),
+            Divide(Column("c"), Literal(integer = Some(5)))))))
+
+    example(
+      query = "SELECT (a % 3 + b) * 2 - c / 5",
+      expectedAst = Project(
+        NoTable(),
+        Seq(
+          Subtract(
+            Multiply(Add(Mod(Column("a"), Literal(integer = Some(3))), Column("b")), Literal(integer = Some(2))),
+            Divide(Column("c"), Literal(integer = Some(5)))))))
+    example(query = "SELECT a || b", expectedAst = Project(NoTable(), Seq(Concat(Column("a"), Column("b")))))
+
+    example(
+      query = "SELECT a || b || c",
+      expectedAst = Project(NoTable(), Seq(Concat(Concat(Column("a"), Column("b")), Column("c")))))
+  }
```

Comment:
These tests would probably be better in a `TSqlExpressionBuilderSpec` and expressed only around the expression part. Like 
```scala
example("a + b", _.expression(), Add(Column("a"), Column("b"))
```

Reply:
Yes - that sounds good. Though I found some issues by using these tests two so I think a combination hits the spot.

Reply:
> Could you elaborate on the issues you faced? Maybe I can help.

To clarify, I mean that going through the SELECT path showed that SELECT without a Table was not currently catered for and I would have missed that if I only went for expression. But I agree that testsing the expression builder independently is the right way. I have no issues understanding the code etc.

Reply:
I think we're on the same page then. We should keep a test for `SELECT 42` (or any query with no `FROM` clause) here and put tests around operator precedence and expressions in general in a `TSqlExpressionBuilderSpec`